docs/knowledge_base/tips/third_party_objects/index.md

```diff
@@ -0,0 +1,170 @@
+On the other hand, the `on_user_content` function is designed to return the HTML content 
+to be rendered, while the `get_user_content_url` function is utilized to obtain the URL 
+for rendering the HTML content.
+
+```python
+html = expose_plotly(sankey_graph(data))
+uc_url = None
+
+def on_user_content(state, path: str, query: dict):
+    return state.html
+
+def on_init(state):
+    state.uc_url = get_user_content_url(state, "val", {"name": "param"})
+```
+
+Finally, we can embed the Sankey Diagram within our web application using the 
+following `part` component:
+
+```
+<|part|page={uc_url}|height=800px|>
+```
+
+This element seamlessly integrates the Sankey Diagram into your web app, providing an 
+engaging user experience.
+
+## Conclusion
+
+Incorporating third-party components into your web applications is a powerful technique 
+that can greatly enhance user engagement. You can achieve this by converting external 
+content into HTML and seamlessly integrating it into your web app as a URL.
+
+This article demonstrated how to embed a Sankey Diagram in your web application using 
+this method. This approach ensures that the integrated content doesn't interfere with 
+your page and provides a secure user experience.
+
+# Entire Code
+
+```python
+# Plotly code to create "fig"
+import plotly.graph_objects as go
+import urllib.request, json
+
+
+url = 'https://raw.githubusercontent.com/plotly/plotly.js/master/test/image/mocks/sankey_energy.json'
+response = urllib.request.urlopen(url)
+data = json.loads(response.read())
+
+
+# override gray link colors with 'source' colors
+opacity = 0.4
+# change 'magenta' to its 'rgba' value to add opacity
+data['data'][0]['node']['color'] = ['rgba(255,0,255, 0.8)' if color == "magenta" else color for color in data['data'][0]['node']['color']]
+data['data'][0]['link']['color'] = [data['data'][0]['node']['color'][src].replace("0.8", str(opacity))
+                                    for src in data['data'][0]['link']['source']]
+
+
+fig = go.Figure(data=[go.Sankey(
+    # Define nodes
+    node = dict(
+      label =  data['data'][0]['node']['label'],
+      color =  data['data'][0]['node']['color']
+    ),
+    # Add links
+    link = dict(
+      source =  data['data'][0]['link']['source'],
+      target =  data['data'][0]['link']['target'],
+      value =  data['data'][0]['link']['value'],
+      label =  data['data'][0]['link']['label'],
+      color =  data['data'][0]['link']['color']
+))])
+
+
+fig.update_layout(title_text="Energy forecast for 2050")
+
+
+# Taipy Code
+from taipy.gui import Gui, get_user_content_url, notify
+import io
+
+
+def expose_plotly(fig):
+    buffer = io.StringIO()
+    fig.write_html(buffer)
+    return buffer.getvalue()
+
+
+def on_user_content(state, path: str, query: dict):
+    return state.html
+
+
+def on_init(state):
+    state.uc_url = get_user_content_url(state, "hello", {"t": "a"})
+
+
+html = expose_plotly(fig)
```

Comment:
Why use this variable?
Why not simply `return expose_plotly(fig)` in *on_user_content()*?